test/orange/detail/BIHMaker.test.cc

```diff
@@ -0,0 +1,120 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file orange/detail/BIHMaker.test.cc
+//---------------------------------------------------------------------------//
+#include "orange/detail/BIHMaker.hh"
+
+#include "corecel/data/CollectionBuilder.hh"
+#include "corecel/data/CollectionMirror.hh"
+#include "celeritas/Types.hh"
+
+#include "celeritas_test.hh"
+
+using BIHMaker = celeritas::detail::BIHMaker;
+
+namespace celeritas
+{
+namespace test
+{
+class BIHMakerTest : public Test
+{
+  public:
+    void SetUp()
+    {
+        auto inf = std::numeric_limits<double>::infinity();
+        bboxes_.push_back({{-inf, -inf, -inf}, {inf, inf, inf}});
+    }
+
+  protected:
+    std::vector<BoundingBox> bboxes_;
+    Collection<LocalVolumeId, Ownership::value, MemSpace::host, OpaqueId<LocalVolumeId>>
+        storage_;
+};
+
+//---------------------------------------------------------------------------//
+//
+/* Simple 4 bbox test. The Z dimensions extends from [0, 100], and it therefore
+ * the longest dimension. However, all bboxes have the same Z-center, so the
+ * first split is X=
+ *
+ *         0    vol 1    1.6
+ *         |--------------|
+ *
+ *                    1.2   vol 2    2.8
+ *                    |---------------|
+ *    y=1 ____________________________________________________
+ *        |           |   |           |                      |
+ *        |           |   |           |      vol 3           |
+ *    y=0 |___________|___|___________|______________________|
+ *        |                                                  |
+ *        |          vols 4, 5 (total overlap)               |
+ *   y=-1 |__________________________________________________|
+ *
+ *        x=0                                                x=5
+ *
+ * Volumes within Resultant tree structure:
+ *
+ *                      root
+ *                   /        \
+ *                 / \      /   \
+ *                1   2    4, 5   3
+ *
+ * 0.8     2     3.9
+ *           2.5
+ */
+
+TEST_F(BIHMakerTest, fourboxes)
+{
+    bboxes_.push_back({{0, 0, 0}, {1.6, 1, 100}});
+    bboxes_.push_back({{1.2, 0, 0}, {2.8, 1, 100}});
+    bboxes_.push_back({{2.8, 0, 0}, {5, 1, 100}});
+    bboxes_.push_back({{0, -1, 0}, {5, 0, 100}});
+    bboxes_.push_back({{0, -1, 0}, {5, 0, 100}});
+
+    BIHMaker bih(bboxes_, &storage_);
+    auto nodes = bih();
+
+    EXPECT_EQ(7, nodes.size());
```

Comment:
Always assert (break early) before operations that would segfault the test if incorrect 😉 
```suggestion
    ASSERT_EQ(7, nodes.size());
```